var/spack/repos/builtin/packages/dramsim2/package.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Dramsim2(MakefilePackage):
+    """
+    DRAMsim is a hardware-validated, cycle-accurate
+    C based simulator for DRAM devices such as DDR3
+    """
+
+    homepage = "https://github.com/umd-memsys/DRAMSim2"
+    git = "https://github.com/umd-memsys/DRAMSim2"
+
+    version('2.2.2',
+            url="https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz",
```

Comment:
Can you move this URL up below the `git` above?

Reply:
Sorry, don't quite understand. Move it out of the `version` call? Or just move it up? It's the only URL right now, but doesn't it make more sense to leave it where it is if it's version-specific?

Reply:
All URLs are version-specific, but Spack applies magic to parse the URL, so you only need to list a single URL for multiple versions:
```console
$ spack url parse https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz
==> Parsing URL: https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz

==> Matched version regex  1: r'^v?(\\d[\\d._-]*)$'
==> Matched  name   regex  0: r'github\\.com/[^/]+/([^/]+)'

==> Detected:
    https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz
                                     --------          ~~~~~
    name:    DRAMSim2
    version: 2.2.2

==> Substituting version 9.9.9b:
    https://github.com/dramninjasUMD/DRAMSim2/archive/v9.9.9b.tar.gz
                                     --------          ~~~~~~
```
So yes, move it out of the `version` call and have it be a top-level variable. This allows things like `spack versions` and `spack checksum` to locate new versions.

Reply:
I made the change. It feels a bit weird to trust that all versions will be located in the same place and with the same file format - but I guess this is rarely an issue and can always be undone with `url=` in the `version` statement (as is done in `intel-pin`).